extensions/elasticsearch-rest-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/lowlevel/runtime/RestClientBuilderHelper.java

```diff
@@ -0,0 +1,67 @@
+package io.quarkus.elasticsearch.restclient.lowlevel.runtime;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.http.HttpHost;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+import org.apache.http.impl.nio.reactor.IOReactorConfig;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestClientBuilder;
+import org.elasticsearch.client.sniff.Sniffer;
+
+public final class RestClientBuilderHelper {
+
+    private RestClientBuilderHelper() {
+        // avoid instantiation
+    }
+
+    static RestClientBuilder createRestClientBuilder(ElasticsearchConfig config) {
+        List<HttpHost> hosts = config.hosts.stream().map(s -> new HttpHost(s.substring(0, s.indexOf(":")),
+                Integer.parseInt(s.substring(s.indexOf(":") + 1)), config.protocol)).collect(Collectors.toList());
```

Comment:
I'm not sure tbh but we should at least change the config stuff to use `InetSocketAddress` IMHO.

Reply:
mongodb, hibernate-search-elasticsearch and neo4j uses such a construct so I think it's OK here.

We can of course make the port optional but it's really standard to define elasticsearch URL this way so I wouldn't bother as it will implies a lot more branches in the configuration code.